### PR TITLE
RntbdServiceEndpoint fix

### PR DESCRIPTION
**Pull request by moderakh**
The intention is to fix this issue:
https://github.com/Azure/azure-sdk-for-java/issues/14094

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java [hunk 1]
```diff
@@ -89,7 +89,10 @@ private RntbdServiceEndpoint(
         this.channelPool = new RntbdClientChannelPool(this, bootstrap, config);
         this.remoteAddress = bootstrap.config().remoteAddress();
         this.concurrentRequests = new AtomicInteger();
-        this.lastRequestNanoTime = new AtomicLong();
+        // if no request has been sent over this endpoint we want to make sure we don't trigger a connection close
+        // due to elapsedTimeInNanos being negative.
+        // idleEndpointTimeoutInNanos - elapsedTimeInNanos <= 0
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java:94]:
```suggestion
        if no request has been sent initially over this endpoint, the below calculation can result in a very big difference 
        final long elapsedTimeInNanos = System.nanoTime() - endpoint.lastRequestNanoTime();
        which can cause endpoint to close unnecessary. 
```

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java]: updated the comment.

**Pull request merged**